### PR TITLE
[Woo POS][Cash & Receipts] Update customer with email and use action to send it

### LINKED ISSUE #13031: [Woo POS][Cash & Receipts] Update customer with email and use action to send it

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #13031

### Description

The PR adds calling of the two endpoints:
* One to update an email of the order
* The second to call action that uses that email to send order details

### Testing information

* Use flipper to monitor network request
* Use the 9.5.0 beta version of the Woo Core
* POS -> Make an order 
* Receipts
* Send a receipt to an email you use
* Validate network calls in flipper

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt [hunk 1]
```diff
@@ -100,6 +105,16 @@ class WooPosTotalsViewModel @Inject constructor(
         }
     }
 
+    private fun sendReceiptByEmail() {
+        val viewState = uiState.value as WooPosTotalsViewState.ReceiptSending
+        val email = viewState.email
+        val orderId = dataState.value.orderId
+        check(orderId != EMPTY_ORDER_ID)
+        viewModelScope.launch {
+            receiptRepository.sendReceiptByEmail(orderId, email)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt:114]: We have no designs, so no handling of the error or success in this PR

Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt:114]: ❓ Do we have plans to validate the email ids here?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this. The API calls are being made as expected. I noticed that we are not validating entered text in the email field which results in making unnecessary API calls that always fails when entered some random text.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt:114]: It depends on the UI, I don't know

**Pull request merged**